File: AlgorithmReview/Programmers/Lv2/택배 상자.swift
//
//  택배 상자.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2023/06/05.
//

import Foundation

func solution(_ order:[Int]) -> Int {
    
    let n = order.max()!
    
    var p = 0
    var container = [Int]()
    var answer = 0
    
    for i in 1...n {
        
        container.append(i)
        
        while p < order.count && container.last == order[p] {
            answer += 1
            container.removeLast()
            p += 1
        }
    }
    
    return answer
}

File: Python/택배 상자.py
def solution(order):
    
    n = max(order)
    p = 0
    container = []
    answer = 0

    for i in range(1,n+1):
        
        container.append(i)
        
        while p < len(order) and len(container) > 0 and order[p] == container[-1]:
            answer += 1
            p += 1
            container.pop()
    
    return answer
